common/src/main/java/com/wynntils/models/items/items/game/EmeraldItem.java

```diff
@@ -6,18 +6,19 @@
 
 import com.wynntils.models.emeralds.type.EmeraldUnits;
 import com.wynntils.models.items.properties.EmeraldValuedItemProperty;
+import java.util.function.Supplier;
 
 public class EmeraldItem extends GameItem implements EmeraldValuedItemProperty {
-    private final int amount;
+    private final Supplier<Integer> amountSupplier;
```

Comment:
I am a bit skeptical to this. Up until now, item annotations have always been immutable. Now the value returned by getAmount() can suddenly start varying from call to call on the same EmeraldItem.

I hear what you say that there are a lot of cases where the item count is changed. But then maybe the actual amount should perhaps not be part of the EmeraldItem at all, but have like a EmeraldModel.getAmount(ItemStack item). It is a bit sad if the EmeraldItem cannot describe its actual value, and just the unit, but if we can't guarantee that it is unchanging, then I think that is a better path to take.

Reply:
(But best would be to actually catch all count changes)